lib/data/repositories/pokemon_repository.dart

```diff
@@ -16,6 +18,27 @@ class PokemonDefaultRepository extends PokemonRepository {
   final GithubDataSource githubDataSource;
   final LocalDataSource localDataSource;
 
+  @override
+  Future<List<Pokemon>> searchPokemon(String pokemonName) async {
+    final hasCachedData = await localDataSource.hasData();
+
+    if (!hasCachedData) {
+      final pokemonGithubModels = await githubDataSource.getPokemons();
+      final pokemonHiveModels = pokemonGithubModels.map((e) => e.toHiveModel());
+
+      await localDataSource.savePokemons(pokemonHiveModels);
+    }
```

Comment:
This one use in multiple places, you should break it into a new method, something like `fetchAndCachePokemons`